src/System.Linq/tests/ConcatTests.cs

```diff
@@ -206,6 +206,16 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             Assert.Throws<OverflowException>(() => supposedlyLargeCollection.Concat(tinyCollection).Count());
             Assert.Throws<OverflowException>(() => tinyCollection.Concat(tinyCollection).Concat(supposedlyLargeCollection).Count());
             Assert.Throws<OverflowException>(() => tinyCollection.Concat(tinyCollection).Concat(tinyCollection).Concat(supposedlyLargeCollection).Count());
+
+            // This applies to ToArray() and ToList() as well, which try to preallocate the exact size
+            // needed if all inputs are ICollections.
+            Assert.Throws<OverflowException>(() => supposedlyLargeCollection.Concat(tinyCollection).ToArray());
+            Assert.Throws<OverflowException>(() => tinyCollection.Concat(tinyCollection).Concat(supposedlyLargeCollection).ToArray());
+            Assert.Throws<OverflowException>(() => tinyCollection.Concat(tinyCollection).Concat(tinyCollection).Concat(supposedlyLargeCollection).ToArray());
+
+            Assert.Throws<OverflowException>(() => supposedlyLargeCollection.Concat(tinyCollection).ToList());
+            Assert.Throws<OverflowException>(() => tinyCollection.Concat(tinyCollection).Concat(supposedlyLargeCollection).ToList());
+            Assert.Throws<OverflowException>(() => tinyCollection.Concat(tinyCollection).Concat(tinyCollection).Concat(supposedlyLargeCollection).ToList());
```

Comment:
All 6 of these asserts fail when run on the current Concat implementation (prior to your src changes), some of them with:
```
Expected: typeof(System.OverflowException)
Actual:   (No exception was thrown)
```
and some with:
```
Expected: typeof(System.OverflowException)
Actual:   typeof(System.OutOfMemoryException): Exception of type 'System.OutOfMemoryException' was thrown.
```
Since these pass with your changes, that suggests the product changes have introduced behavioral changes.